Lind-Project/lind_project
# Test files for sync_file_range

## Description
This PR introduces two new test cases for the sync_file_range syscalls. The test files have been created and tested for validity. They will help us confirm the correct operation of these syscalls.


### sync_file_range.c
We've introduced a C source file, `sync_file_range.c`, to verify the `sync_file_range` system call. 
### Type of change

## How Has This Been Tested?
To reproduce the tests, follow these steps for each system call:

### Test of sync_file_range
- Test of `sync_file_range`- `lind_project/tests/test_cases/sync_file_range.c`
1. Compile `sync_file_range.c` in `lindsh` with `compile tests/test_cases/sync_file_range.c -o sync_file_range.nexe`
2. Copy test file to Lind file system with `lindfs cp /home/lind/lind_project/sync_file_range.nexe sync_file_range.nexe`
3. If all operations were successful, your file should have been copied. You can check the file with `lindfs ls /` command.
4. Run the test file with `lind /sync_file_range.nexe`

## Checklist:



- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, safeposix-rust)
